Flagging

pondorasti opened the pull request:
## Flagging Logic

- created FlaggingContentItems protocol to store the flaggable items for each viewController
- the FlaggingViewController is alway triggered by an action sheet

## FlaggingViewController

- this screen should be presented in a inside navigation controller modally
- finished conforming to the tableView protocols

## Refactored the following classes, regarding the names & marks

- FlaggingViewController
- StatusPagerTabStripViewController

## Notes

- I feel that the FlaggingViewController initializer is a bit redundant since it asks for user to report and also donation to report(hence we could get the user from the donation object), didn't modify anything here and left out the old implementation
- The flagging logic in StatusPagerTabStripViewController should be double checked to make sure we are flagging the right person (might have made a little mistake here)
- We are missing the Flagging Item titles
- Inside the ViewController there I have left a todo for using the FlaggingService
- I didn't have the chance to test anything

Commit:
WIP: FlaggingViewController

+ Refactored and added marks
+ UITableViewDelegate protocol function implementation

Commit:
WIP: Flagging

+ FlaggingContentItems Protocol
+ finished implementing the flagging logic for DetailedDonationModularTableViewController
+ Fixed ModularTableView scrollView bottom offset

Commit:
Done: flagging logic for VerifyDropoffViewController

Commit:
Refactored StatusPagerTabStripViewController

Commit:
Flagging Logic Implementation for StatusPagerTabStripViewController

LinnierGames commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Flagging](https://trello.com/c/d4WjFYpK/150-flagging)

LinnierGames committed:
Use ReportingService to Create the Report

LinnierGames committed:
User Friendly Strings, Small fixes

Removed unnecessary user/donation in the FlaggingVc init

LinnierGames committed:
Send Reports for Donations

when a donation is reported, all copies of that donation are updated

LinnierGames commented on KifuSF/DetailedDonationModularTableViewController.swift:
```diff
@@ -42,7 +42,26 @@ class DetailedDonationModularTableViewController: ModularTableViewController {
+            title: "Report Donation",
+            style: .default) { (_) in
+                self.present(flaggingViewController, animated: true)
+        }
```
are we only reporting a donation here? if so, there is no need to pass in a `user` in the constructor of the FlaggingVc

pondorasti replied:
Agree! I wasn't sure how the FlaggingService worked, so I had the urge to use both constructor parameters :]

LinnierGames commented on KifuSF/FlaggingViewController.swift:
```diff
@@ -0,0 +1,177 @@
+        
+        let selectedItem = flaggableItems[indexPath.row]
+        
+        //TODO: ask the user for a message on why they flagged it
```
TODO: this is where the message is added to the report

pondorasti replied:
done!

LinnierGames replied:
Great!

LinnierGames commented on KifuSF/VerifyDropoffViewController.swift:
```diff
@@ -89,11 +89,23 @@ class VerifyDropoffViewController: UIScrollableViewController {
+                donationToReport: donation
+            )
+        )
         navigationController?.pushViewController(flaggingViewController, animated: true)
```
I thought you mentioned to present the FlaggingVc modally?

pondorasti replied:
The rootViewController of that nav controller is already presented modally(which is the VerifyDropoffViewController), so I thought it's better to push this screen rather than presenting it.

LinnierGames replied:
Sounds reasonable

LinnierGames commented on KifuSF/Report.swift:
```diff
@@ -21,18 +21,28 @@ enum FlaggedContentType: Int {
     case flaggedCommunication
     
-    func getDescription() -> String {
+    var description: String {
```
tell me what you think about this change?

pondorasti replied:
Computed property looks like the way to go for this case

LinnierGames commented on KifuSF/Report.swift:
```diff
@@ -21,18 +21,28 @@ enum FlaggedContentType: Int {
+        case .flaggedPhoneNumber:
+            return "Incorrect Phone Number"
+        case .flaggedCommunication:
+            return "Inapropiate/Unresponsive Communication"
```
Adjust the string values to what you feel is best, please

pondorasti replied:
done!

LinnierGames reviewed (changes requested):
This looks great! I've added many comments but only one I am requesting a change: allow the user to add a message to the report.

LinnierGames committed:
Merge branch 'dev' into flagging

Commit:
Update FlaggedContentType description

pondorasti commented on KifuSF/StatusPagerTabStripViewController.swift:
```diff
@@ -9,119 +9,90 @@
-            donationVc.donation = donation
-        }
+extension StatusPagerTabStripViewController: FlaggingContentItems {
+    var flaggableItems: [FlaggedContentType] {
```
I have just noticed that theoretically, we have the functionality to flag your own donation post.

pondorasti replied:
I have added new logic for handling logic, that you should go over it just to double check everything.

Commit:
Better logic for handling flagging in StatusPagerTabStripViewController

Commit:
FlaggingViewController logic for retrieving comment

Commit:
Better Flagging Logic for StatusPagerTabStripViewController

LinnierGames committed:
Minor Fixes

Flaggable item codes were incorrect. The defer method was never executed when a report is nil

LinnierGames commented:
In the `Minor Fixes` commit, the `defer` block of code was not executed when the report was nil. This is because for the defer block to work execution needs to *reach* that block. That's why I moved the `defer` before the guard statement.

Do I make sense??
